v4l2loopback.c

```diff
@@ -2552,7 +2552,7 @@ static int v4l2_loopback_add(struct v4l2_loopback_config *conf, int *ret_nr)
 	dprintk("buffer_size = %ld (=%d)\n", dev->buffer_size,
 		dev->pix_format.sizeimage);
 	err = allocate_buffers(dev);
-	if (err && dev->buffer_size)
+	if (err < 0)
```

Comment:
this fails in the default case (where the image is 0x0, and thus `dev->buffer_size == 0`